macros/get_metafields.sql

```diff
@@ -0,0 +1,49 @@
+{% macro get_metafields(source_object, reference_value, lookup_object="stg_shopify__metafield", key_field="metafield_reference", key_value="value", reference_field="owner_resource") %}
+
+{% set pivot_fields = dbt_utils.get_column_values(table=ref(lookup_object), column=key_field, where="lower(" ~ reference_field ~ ") = lower('" ~ reference_value ~ "')") %}
+
+{% set source_columns = adapter.get_columns_in_relation(ref(source_object)) %}
+{% set source_column_count = source_columns | length %}
+
+with source_table as (
+    select *
+    from {{ ref(source_object) }}
+)
+
+{% if pivot_fields is not none %},
+lookup_object as (
+    select 
+        *,
+        {{ dbt_utils.pivot(
+                column=key_field, 
+                values=pivot_fields, 
+                agg='', 
+                then_value=key_value, 
+                else_value="null",
+                quote_identifiers=false
+                ) 
+        }}
+    from {{ ref(lookup_object) }}
+),
+
+final as (
+    select
+        source_table.*
+        {% for fields in pivot_fields %}
+            , max(replace(replace(lookup_object.{{ dbt_utils.slugify(fields) }},'["',''),'"]','')) as metafield_{{ dbt_utils.slugify(fields) }}
```

Comment:
should we be doing the `replace` for square brackets if we're not parsing out other kinds of formats? 

if the answer is still yes, are we trying to get rid of the brackets at just the beginning and end of the field? if so, i think `trim` might be safer than `replace`

Reply:
This is a good question. I wanted to replace those because they were included for every type of format (with the exception of the json fields). So the metafield value would be presented as `["5"]`, which is not useful unless the customer builds another model on top of ours that replaces those square branches and quotes. This was my attempt to make the metafield models a bit more useful out of the box.

I would much rather use TRIM, however if I recall it is not supported on all warehouses 😢. So it was either make a complex macro for fivetran_utils or use replace. Let me try and use TRIM and see if we are in the clear.

Reply:
So it turns out the RTRIM and LTRIM (which would be the safer options here) don't exist on all warehouses. However, TRIM does exist to some extent. Although TRIM is a bit more dangerous in my opinion as it will TRIM **any** instance of the characters. So `trim('"[}]')` would actually remove any bracket or quote which is not at all what we want. 

Therefore, I think we should either keep the replace, or completely remove it. I am leaning to keeping the replace for the first release and seeing how customers receive it. If it causes issues downstream then we can readdress it. What are your thoughts?

Reply:
As discussed during standup we will remove the replace.

Reply:
Removed